pkg/platform/runtime/runtime.go

```diff
@@ -152,8 +146,15 @@ func (r *Runtime) validateCache() error {
 		expr = string(data)
 	}
 
-	if r.cfg.GetBool(constants.OptinBuildscriptsConfig) && !script.EqualsBuildExpressionBytes([]byte(expr)) {
-		return &NeedsCommitError{errs.New("Runtime changes should be committed")}
+	// Check if local build script has changes that should be committed.
+	if r.cfg.GetBool(constants.OptinBuildscriptsConfig) {
+		script, err := buildscript.NewScriptFromProject(r.target, r.auth)
+		if err != nil {
+			return errs.Wrap(err, "Unable to get local build script")
+		}
```

Comment:
My initial thought was that it's creating a build script from the just-stored buildexpression and comparing that expr against itself (which will never fail). The original intent of this block being above was to ensure using the previous build expression, and not the new one.

I turned on the "Test: commit" label to try and catch this. It failed, but I don't know if it was because of this concern though. Regardless, that failure needs to be addressed first.